lib/puppet/provider/package/pkgng.rb

```diff
@@ -0,0 +1,145 @@
+require 'puppet/provider/package'
+
+Puppet::Type.type(:package).provide :pkgng, :parent => Puppet::Provider::Package do
+  desc "A PkgNG provider for FreeBSD and DragonFly."
+
+  commands :pkg => "/usr/local/sbin/pkg"
+
+  confine :operatingsystem => [:freebsd, :dragonfly]
+  confine :pkgng_enabled => :true
+
+  defaultfor :operatingsystem => :freebsd
+  defaultfor :pkgng_enabled => :true
```

Comment:
In plain `rspec` (as opposed to `rspec-puppet`), you cannot stub in fact values through `let(:facts)`.

There are [examples](https://github.com/puppetlabs/puppet/blob/6500619c6e26cbef26faf4ee45bdedc9e70f4690/spec/unit/provider_spec.rb#L291) in the spec suite on how to do it instead. Let us know if you require more information.